contrib/node/src/python/pants/contrib/node/subsystems/resolvers/npm_resolver.py

```diff
@@ -30,17 +30,46 @@ def register_options(cls, register):
     register(
       '--install-optional', type=bool, default=False, fingerprint=True,
       help='If enabled, install optional dependencies.')
+    register(
+      '--production-only', type=bool, default=False, fingerprint=True,
+      help='If enabled, only install production dependencies')
+    register(
+      '--force', type=bool, default=False, fingerprint=True,
+      help='If enabled, re-download dependencies.')
+    register(
+      '--frozen-lockfile', type=bool, default=True, fingerprint=True,
+      help='If enabled, disallow automatic update of lock files.')
+    register(
+      '--force-option-override', type=bool, default=False, fingerprint=True,
+      help='ADVANCED option. If enabled, options will override hard-coded values.')
```

Comment:
There is an `advanced=True` argument to register, which hides options from `./pants help` and only shows them for `./pants help --advanced`.

But it's not clear what this means... probably needs an example.